custom/icds_reports/utils/__init__.py

```diff
@@ -1729,6 +1733,29 @@ def phone_number_function(x):
     return "+{0}{1}".format('' if str(x).startswith('91') else '91', x) if x else x
 
 
+def filter_cas_data_export(export_file, location):
+    with TransientTempfile() as path:
+        with open(path, 'wb') as temp_file:
+            temp_file.write(export_file.get_file_from_blobdb().read())
+        with open(path, 'r') as temp_file:
+            fd, export_file_path = mkstemp()
+            csv_file = os.fdopen(fd, 'w')
+            reader = csv.reader(temp_file)
+            writer = csv.writer(csv_file)
+            headers = next(reader)
+            for i, header in enumerate(headers):
+                if header == f'{location.location_type.name}_name':
+                    index_of_location_type_name_column = i
+                    break
+            else:
+                raise InvalidLocationType(f'{location.location_type.name} is not a valid location option for cas data exports')
```

Comment:
Neat I have never heard of or used `for ... else` before.

Reply:
I don't love using somewhat unusual syntax but it did feel like a perfect fit for this instance